docs/DeveloperGuide.md

```diff
@@ -0,0 +1,125 @@
+# RepoSense - Developer Guide
+Thank you for your interest in contributing to RepoSense!
+- [Setting up](#setting-up)
+  - [Prerequisites](#prerequisites)
+  - [Setting up the project in your computer using IntelliJ](#setting-up-the-project-in-your-computer-using-intellij)
+  - [Verifying the setup](#verifying-the-setup)
+  - [Configuring the coding style](#configuring-the-coding-style)
+- [Architecture](#architecture)
+  - [Parser](#parserconfigparser)
+  - [Git](#gitgitdownloader)
+  - [CommitsReporter](#commitsreporter)
+  - [AuthorshipReporter](#authorshipreporter)
+  - [ReportGenerator](#reportgeneratormain)
+  - [System](#system)
+  - [Model](#model)
+- [HTML Dashboard](#html-dashboard)
+
+## Setting up
+
+### Prerequisites
+1. **JDK `1.8.0_60`** or later.
+2. **git `2.14`** or later on the command line.
+ * Check that this tool exist on your OS terminal by typing its name on the terminal and ensure that the terminal does not output messages such as `not found` or `not recognized`.
+
+### Setting up the project in your computer using IntelliJ
+1. Fork this repo, and clone the fork to your computer.
+2. Open *IntelliJ* (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct *JDK* version for *Gradle*.
+4. Click `Configure` > `Project Defaults` > `Project Structure`.
+5. Click `New…​` and find the directory of the *JDK*.
```

Comment:
Steps 4 and 5 are sub-steps of 3? If not, explain how to do step 1 (unless it is trivial)?
BTW, we can use `1.` in each line instead of specifying the item number. Markdown will insert the correct numbers. That way, it is easier to insert steps in future.